db/db_impl_write.cc

```diff
@@ -584,6 +716,38 @@ Status DBImpl::WriteToWAL(const WriteThread::WriteGroup& write_group,
   return status;
 }
 
+Status DBImpl::ConcurrentWriteToWAL(const WriteThread::WriteGroup& write_group,
```

Comment:
It is actually concurrent. With new changes we could have two threads trying to write to the WAL at the same time: the thread aggregating prepares and the thread writing commit.
Not flushing WAL buffer is a totally separate feature. It happen to be in the same patch since these two features together give performance boost.